src/librustc_mir/dataflow/framework/cursor.rs

```diff
@@ -29,32 +30,32 @@ where
 
     pos: CursorPosition,
 
-    /// When this flag is set, the cursor is pointing at a `Call` or `Yield` terminator whose call
-    /// return or resume effect has been applied to `state`.
+    /// Indicates that `state` has been modified with a custom effect.
     ///
-    /// This flag helps to ensure that multiple calls to `seek_after_assume_success` with the
-    /// same target will result in exactly one invocation of `apply_call_return_effect`. It is
-    /// sufficient to clear this only in `seek_to_block_start`, since seeking away from a
-    /// terminator will always require a cursor reset.
-    success_effect_applied: bool,
+    /// When this flag is set, we need to reset to an entry set before doing a seek.
+    state_needs_reset: bool,
 }
 
 impl<'mir, 'tcx, A, R> ResultsCursor<'mir, 'tcx, A, R>
 where
     A: Analysis<'tcx>,
     R: Borrow<Results<'tcx, A>>,
 {
-    /// Returns a new cursor for `results` that points to the start of the `START_BLOCK`.
+    /// Returns a new cursor for `results` that points to the fixpoint of the `START_BLOCK`.
     pub fn new(body: &'mir mir::Body<'tcx>, results: R) -> Self {
         ResultsCursor {
             body,
-            pos: CursorPosition::BlockStart(mir::START_BLOCK),
-            state: results.borrow().entry_sets[mir::START_BLOCK].clone(),
-            success_effect_applied: false,
+            pos: CursorPosition::fixpoint(mir::START_BLOCK),
```

Comment:
@ecstatic-morse did you overlook the case here?

Or are you implicitly arguing that the term "fixpoint" is appropriate here? (A potential viewpoint which I have attempted to understand in a comment elsewhere, but it isn't a viewpoint that I have been convinced to adopt...)

Reply:
I missed this case. Ugh.

Reply:
Ah this is just in a commit. Not the final PR. Presumably missed by git absorb

Reply:
(it seems like it must have been deleted by a later commit, now that I've compared against a full diff...)

anyway, thanks for confirming!